icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/PersonNameFormatterTest.java

```diff
@@ -355,22 +364,34 @@ public void TestNameSpacing() {
             // the native formatter
             new NameAndTestCases("locale=ja_JP,given=駿,surname=宮崎", new String[][] {
                 { "ja_JP", "LONG",   "REFERRING",  "FORMAL",   "DEFAULT", "", "宮崎駿" },
-                { "zh_CN", "LONG",   "REFERRING",  "FORMAL",   "DEFAULT", "", "宮崎 駿" },
+                { "zh_CN", "LONG",   "REFERRING",  "FORMAL",   "DEFAULT", "", "宮崎·駿" },
             }),
 
             // if the formatter locale doesn't use spaces and the name's locale does, use the name locale's formatter,
             // but if the name is still using the formatter locale's script, use the native formatter's
             // "foreign space replacement" character instead of spaces
             new NameAndTestCases("locale=en_US,given=Albert,surname=Einstein", new String[][] {
-                { "ja_JP", "LONG",   "REFERRING",  "FORMAL",   "DEFAULT", "", "Albert Einstein" },
-                { "zh_CN", "LONG",   "REFERRING",  "FORMAL",   "DEFAULT", "", "Albert Einstein" },
+// NOTE TO REVIEWERS: These two tests fail with "Albert・Einstein", which is clearly wrong.  The files in
```

Comment:
Good catch.

I think what is happening is if the name is in a different script, it formats with a different locale's formatter. That is in the LDML spec but not in CLDR's tooling, because for gathering data we don't need that.

So you need to add the logic to:
1. get the script of a PersonName
2. reroute PersonNames with a different script to a different locale.

I think this is clear in the spec, but if not we can discuss.